SPEC.md

```diff
@@ -9481,6 +9482,72 @@ Example output:
 </p>
 </details>
 
+### ✨ `chunk`
+
+```
+Array[Array[X]] chunk(Array[X], Int)
+```
+
+Given an array and a length *n*, splits the array into consecutive, non-overlapping arrays of *n* elements. If the length of the array is not a multiple *n* then the final sub-array will have `length(array) % n` elements.
+
+This function is the inverse of [`flatten`](#flatten); i.e., `flatten`ing a `chunk` array will always yield the original array.
+
+**Parameters**
+
+1. `Array[X]`: The array to split. May be empty.
+2. `Int`: The desired length of the sub-arrays. Must be > 0.
+
+**Returns**: An array of sub-arrays, where each sub-array is the same length except possibly the last one.
+
+<details>
+<summary>
+Example: chunk_array.wdl
+
+```wdl
+version 1.2
+workflow chunk_array {
+  Array[String] s1 = ["a", "b", "c", "d", "e", "f"]
+  Array[String] s2 = ["a", "b", "c", "d", "e"]
+  Array[String] s3 = ["a", "b"]
+  Array[String] s4 = []
+  
+  scatter (a in split(s1, 3)) {
```

Comment:
Not sure if this scatter block is needed to demonstrate the chunk function?

Reply:
I'm open to suggestions if you think another example would be better.